### DISCUSSION TITLE
ファームウェアが書き込めない

### DISCUSSION THREAD

**Discussion by kawamataryo** [Yowkees/keyball#594] [Q&A]
自作キーボード初心者です。Keyball44を購入してビルドガイドにしたがって組み立てを行ったのですが、[ProMicroへのファームウェアの書き込み](https://github.com/Yowkees/keyball/blob/main/keyball44/doc/rev1/buildguide_jp.md#3-8promicro%E3%81%AE%E6%9B%B8%E3%81%8D%E8%BE%BC%E3%81%BF%E3%81%A8%E5%8B%95%E4%BD%9C%E7%A2%BA%E8%AA%8D)で、ProMicroが認識されず困っています。

解決方法や、確認箇所などございましたらご教授お願いします🙏

### 状況
- REMAPに接続して、FLASH（caterina）を選択しても接続デバイスにProMicroが表示されない

### 環境
- macOS sonoma 14.4.1
- Keyball44
- [ProMicro typeC](https://shop.yushakobo.jp/products/3905)

### その他情報
- 左右ともに初回接続時に一瞬認識され書き込みがスタートしたが、その後エラーで止まった（そのときのエラーの内容はメモし忘れました・・）
　- その後、リトライするもcaterinaのデバイス一覧に表示されず
- RESETスイッチを素早く2回押しても状況は変わらず
- FLASHでbootloaderをdfuとするとデバイス一覧にkeyball44と表示される
  - ただ、その状態で書き込みをしてもエラーになる （`Error: Unsupported device: Vendor ID: 5957 Product ID: 400`）
- PCに接続するとMacのキーボードアシスタントが起動する
- キースイッチをピンセットでショートしてもテキストエディタに文字は表示されない
- `$system_profiler SPUSBDataType`を実行すると表示される

```
    USB 3.1 Bus:

      Host Controller Driver: AppleT6000USBXHCI

        Keyball44:

          Product ID: 0x0400
          Vendor ID: 0x5957
          Version: 1.00
          Speed: Up to 12 Mb/s
          Manufacturer: Yowkees
          Location ID: 0x00100000 / 1
          Current Available (mA): 500
          Current Required (mA): 500
          Extra Operating Current (mA): 0
```

**Comment by Njima1572** [comment]
同じ症状かどうかは分からず参考になるか分かりませんが、自分も同じくRemapでFirmwareの書き込みをしようとした際、Unsupported deviceのエラーに遭遇しました。

自分の環境では、remapからダウンロードしたhexファイルを用いて
```
qmk flash -m ATmega32U4 -bl avrdude ./Keyball44-1711413474675.hex
```
で書き込みすることが出きました。(特にどのポートにProMicroが差さっているのかなどは指定していないのに勝手にみつけてくれてました。)

**Reply by kawamataryo** [reply]
おお！ありがとうございます🙇
今日そちらの方法を試してみます

**Comment by yukiyohure** [comment]
私も同じ現象が起こっていたのですが、REMAPを開くブラウザをBraveからChromeに変更したらbootloaderをcaterinaでファームウェアの書き込みが完了しました。
参考になればと思います。

**Comment by kawamataryo** [comment]
片方はリセットボタンの2度押しの間隔調整、片方はGRD, RSTのピンセットでのショート（リセットのハンダが悪かった模様）で、ファームウェアの書き込みまではできました！

ファームウェアは書き込めたものの動かないので、まだアレですが一旦このDiscussionはクローズします。

回答いただいた @Njima1572 さん、@yukiyohure さん本当にありがとうございます！！🙏
